File: backend/src/main/java/com/PIDEV/demo/entities/Claim.java
package com.PIDEV.demo.entities;

import lombok.*;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@ToString
public class Claim implements Serializable {

    private static final long serialVersionUID = 1L;
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private Integer idClaim;
    @Temporal(TemporalType.DATE)
    private Date dateC;
    private String descriptionC;
    private String productType;
    private String claimType;
    private Boolean etat =false;

    @ManyToOne(cascade = CascadeType.ALL)
    private Delivery delivery;

    @ManyToOne(cascade = CascadeType.ALL)
    private Customer customer;

    @ManyToOne(cascade = CascadeType.ALL)
    private CommandLigne CommandLigness;


    @ManyToOne
    private Publication publication;


}

File: frontend/src/app/Models/claim.ts
export class Claim {
    idClaim  : Number;
    dateC : Date;
    descriptionC : String;
    productType: String;
    claimType:String;
    etat: boolean;
    delivery: Number;
    customer: Number;
    CommandLigness: Number;
    publication: Number;

    constructor(){
        this.idClaim=0;
        this.dateC= new Date();
        this.descriptionC="";
        this.productType="";
        this.claimType="";
        this.etat=false;
        this.delivery=0;
        this.customer=0;
        this.CommandLigness=0;
        this.publication=0;

    }
    
}
